cmd/mdatagen/internal/metadata.go

```diff
@@ -270,6 +277,56 @@ func validateEvents(events map[EventName]Event, attributes map[AttributeName]Att
 	return errs
 }
 
+func (md *Metadata) validateFeatureGates() error {
```

Comment:
Can we also validate the following:
1. The feature gates are sorted
2. Each feature gate has a link
3. The id follows the regular expression `^[0-9a-zA-Z.]*$` (see https://github.com/open-telemetry/opentelemetry-collector/blob/a6a176660925fd2826ba83d64910f05a2e365235/featuregate/registry.go#L122-L124)